public_html/wp-content/themes/pattern-directory/src/components/my-favorites/index.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as patternStore } from '../../store';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import PatternGrid from '../pattern-grid';
+import PatternGridMenu from '../pattern-grid-menu';
+import PatternThumbnail from '../pattern-thumbnail';
+import QueryMonitor from '../query-monitor';
+import { RouteProvider } from '../../hooks';
+
+const MyFavorites = () => {
+	const { favorites, isLoggedIn, query } = useSelect( ( select ) => ( {
+		favorites: select( patternStore ).getFavorites(),
+		isLoggedIn: !! select( coreStore ).getCurrentUser()?.id,
+		query: select( patternStore ).getCurrentQuery(),
+	} ) );
+
+	if ( ! isLoggedIn ) {
```

Comment:
This is flashing on page load, can we delay its rendering until after we are certain the user is not logged in? Same goes for the `You haven’t favorited any patterns yet` view.

![](https://d.pr/i/sSL1y1.gif)

Reply:
There doesn't seem to be a loading state for users, `currentUser` is `{}` whether waiting or not logged in.

Reply:
Updated this to use a value provided by the server, so there's no user loading state anymore… but now it shows `You haven’t favorited any patterns yet` until _that_ loads. Will see if I can address that as part of #280.